### PR TITLE
Consolidate with g8s efk

### PR DESCRIPTION
**Pull request by QuentinBisson**
Towards https://github.com/giantswarm/giantswarm/issues/21581

Tested on Gauss

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -6,12 +6,98 @@ workflows:
   package-and-push-chart-on-tag:
     jobs:
       - architect/push-to-app-catalog:
-          name: "package and push chart to app catalog"
-          app_catalog: "giantswarm-playground-catalog"
-          app_catalog_test: "giantswarm-playground-test-catalog"
+          name: push-to-giantswarm-catalog
```
Comment (QuentinBisson) [inline .circleci/config.yml:9]: Fix circle ci to deploy in app collections

File: helm/fluent-logshipping-app/Configuration.md [hunk 2]
```diff
@@ -29,7 +28,6 @@ The following table lists the configurable parameters of the fluent-logshipping-
 | `outputs.aws.S3.bucketName`                           | outputs S3 plugin bucket name                                                     | `my-cluster-logs` |
 | `outputs.aws.S3.bucketRegion`                         | outputs S3 plugin region                                                          | `eu-central-1`    |
 | `outputs.aws.S3.bucketPathPrefix`                     | outputs S3 plugin prefix                                                          | `gs-`             |
-| `outputs.aws.S3.account`                              | outputs S3 plugin default account of the S3 bucket                                | `0000000000`      |
```
Comment (QuentinBisson) [inline helm/fluent-logshipping-app/Configuration.md:32]: Value was not used

File: helm/fluent-logshipping-app/templates/configmap.yaml [hunk 3]
```diff
@@ -40,9 +41,15 @@ data:
     @INCLUDE outputs.conf
 
   inputs.conf: |
+    {{- if has "containers" .Values.outputs.inputLogTypes }}
```
Comment (QuentinBisson) [inline helm/fluent-logshipping-app/templates/configmap.yaml:44]: configure inputs only if the input log type is configured in the app

File: helm/fluent-logshipping-app/templates/configmap.yaml [hunk 4]
```diff
@@ -67,27 +74,27 @@ data:
...
     [INPUT]
         Name              systemd
         Tag               audit.ssh.*
-        Path              /run/log/journal
-        Parser            syslog
         DB                ${input_database_path}/audit_ssh.db
-        Systemd_Filter    SYSLOG_IDENTIFIER=sshd
-        Mem_Buf_Limit     ${input_memory_buffer_limit}
         storage.type      {{ .Values.fluentbit.inputStorageTypes.sshd }}
+        Mem_Buf_Limit     ${input_memory_buffer_limit}
```
Comment (QuentinBisson) [inline helm/fluent-logshipping-app/templates/configmap.yaml:91]: Reorganizing to match the others

File: helm/fluent-logshipping-app/templates/configmap.yaml [hunk 5]
```diff
@@ -67,27 +74,27 @@ data:
...
         Strip_Underscores on
+        Systemd_Filter    SYSLOG_IDENTIFIER=sshd
```
Comment (QuentinBisson) [inline helm/fluent-logshipping-app/templates/configmap.yaml:93]: Supporting both syslog_identifier and systemd_unit. This is an OR relationship by default

File: helm/fluent-logshipping-app/templates/configmap.yaml [hunk 6]
```diff
@@ -128,6 +136,46 @@ data:
         Match  *
         Record cluster {{ .Values.clusterID }}
 
+  parsers.conf: |
```
Comment (QuentinBisson) [inline helm/fluent-logshipping-app/templates/configmap.yaml:139]: Moved the parser up in the conf file

File: helm/fluent-logshipping-app/templates/daemonset.yaml [hunk 7]
```diff
@@ -1,24 +1,24 @@
+{{- if and .Values.outputs .Values.outputs.inputLogTypes }}
```
Comment (QuentinBisson) [inline helm/fluent-logshipping-app/templates/daemonset.yaml:1]: This is used to not create the components if the no input logs are configured

File: .circleci/config.yml [hunk 8]
```diff
@@ -6,12 +6,98 @@ workflows:
...
+          name: push-to-control-plane-catalog
+          app_catalog: "control-plane-catalog"
+          app_catalog_test: "control-plane-test-catalog"
+          chart: "fluent-logshipping-app"
+          filters:
+            tags:
+              only: /^v.*/
+
+      - architect/push-to-app-collection:
+          name: "push to-shared-app-collection"
```
Comment (TheoBrigitte) [inline .circleci/config.yml:28]:
```suggestion
          name: "push-to-shared-app-collection"
```

File: .circleci/config.yml [hunk 9]
```diff
@@ -6,12 +6,98 @@ workflows:
...
+          requires:
+            - push-to-control-plane-catalog
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /^v.*/
+
+      - architect/push-to-app-collection:
+          name: "push to-aws-app-collection"
```
Comment (TheoBrigitte) [inline .circleci/config.yml:41]:
```suggestion
          name: "push-to-aws-app-collection"
```

File: .circleci/config.yml [hunk 10]
```diff
@@ -6,12 +6,98 @@ workflows:
...
+          requires:
+            - push-to-control-plane-catalog
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /^v.*/
+
+      - architect/push-to-app-collection:
+          name: "push to-azure-app-collection"
```
Comment (TheoBrigitte) [inline .circleci/config.yml:54]:
```suggestion
          name: "push-to-azure-app-collection"
```

File: .circleci/config.yml [hunk 11]
```diff
@@ -6,12 +6,98 @@ workflows:
...
+          requires:
+            - push-to-control-plane-catalog
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /^v.*/
+
+      - architect/push-to-app-collection:
+          name: "push to-kvm-app-collection"
```
Comment (TheoBrigitte) [inline .circleci/config.yml:67]:
```suggestion
          name: "push-to-kvm-app-collection"
```

File: .circleci/config.yml [hunk 12]
```diff
@@ -6,12 +6,98 @@ workflows:
...
+          requires:
+            - push-to-control-plane-catalog
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /^v.*/
+
+      - architect/push-to-app-collection:
+          name: "push to-vsphere-app-collection"
```
Comment (TheoBrigitte) [inline .circleci/config.yml:80]:
```suggestion
          name: "push-to-vsphere-app-collection"
```

File: .circleci/config.yml [hunk 13]
```diff
@@ -6,12 +6,98 @@ workflows:
...
+          requires:
+            - push-to-control-plane-catalog
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /^v.*/
+      
+      - architect/push-to-app-collection:
+          name: "push to-openstack-app-collection"
```
Comment (TheoBrigitte) [inline .circleci/config.yml:93]:
```suggestion
          name: "push-to-openstack-app-collection"
```

File: helm/fluent-logshipping-app/templates/configmap.yaml (hunk 5 above)
Comment (TheoBrigitte) [reply helm/fluent-logshipping-app/templates/configmap.yaml:93]: Maybe add this as a comment in the file.

File: helm/fluent-logshipping-app/values-s3.yaml [hunk 14]
```diff
@@ -1,12 +0,0 @@
...
-  aws:
-    kiam: true
-    S3:
-      enabled: true
-      bucketName: "my-cluster-logs"
-      bucketRegion: "eu-central-1"
-      bucketPathPrefix: "gs-"
-      account: "00000000000"
-      role: "my-role"
-      s3_object_key_format: "%{path}%{time_slice}_%{index}.%{file_extension}"
```
Comment (TheoBrigitte) [inline helm/fluent-logshipping-app/values-s3.yaml:11]: Why removing this? It seems like a good example, of how to use those values.

**Review by TheoBrigitte — CHANGES_REQUESTED** [review]

File: helm/fluent-logshipping-app/values-s3.yaml (hunk 14 above)
Comment (QuentinBisson) [reply helm/fluent-logshipping-app/values-s3.yaml]: I added it back :)

**Comment by QuentinBisson** [comment]
@TheoBrigitte I addressed your review comment

File: helm/fluent-logshipping-app/templates/configmap.yaml [hunk 15]
```diff
@@ -67,27 +74,28 @@ data:
...
         Tag               audit.ssh.*
-        Path              /run/log/journal
-        Parser            syslog
         DB                ${input_database_path}/audit_ssh.db
-        Systemd_Filter    SYSLOG_IDENTIFIER=sshd
-        Mem_Buf_Limit     ${input_memory_buffer_limit}
         storage.type      {{ .Values.fluentbit.inputStorageTypes.sshd }}
+        Mem_Buf_Limit     ${input_memory_buffer_limit}
         Strip_Underscores on
+        ## Filters journalctl logs for either the syslog identifier=sshd of the _SYSTEMD_UNIT=sshd.service to ensure we only get SSH logs.
```
Comment (TheoBrigitte) [inline helm/fluent-logshipping-app/templates/configmap.yaml:93]: I though this was an OR operator?

```suggestion
        ## Filters journalctl logs either by syslog identifier or systemd unit name to ensure we only get SSH logs.
```

File: helm/fluent-logshipping-app/values-s3.yaml [hunk 16]
```diff
@@ -1,12 +1,10 @@
...
+    role: arn:aws:iam::00000000000:role/my-role
     S3:
       enabled: true
       bucketName: "my-cluster-logs"
-      bucketRegion: "eu-central-1"
-      bucketPathPrefix: "gs-"
-      account: "00000000000"
-      role: "my-role"
+      totalFileSize: "100M"
       s3_object_key_format: "%{path}%{time_slice}_%{index}.%{file_extension}"
```
Comment (TheoBrigitte) [inline helm/fluent-logshipping-app/values-s3.yaml:10]: Nice, thanks.

Comment (TheoBrigitte) [reply helm/fluent-logshipping-app/values-s3.yaml:10]: We can still specify `bucketPathPrefix` right?

**Pull request merged**